Trimble.Modus.Components/Controls/TMSegmentedControl/TMSegmentedControl.xaml.cs

```diff
@@ -0,0 +1,499 @@
+    /// <param name="newValue">The new value of the enabled state.</param>
+    private static void OnEnabledStateChanged(BindableObject bindable, object oldValue, object newValue)
+    {
+        if (bindable is TMSegmentedControl segmentedControl)
+        {
+            segmentedControl.Opacity = segmentedControl.IsEnabled ? 1 : 0.5;
+            foreach (TMSegmentedItem item in segmentedControl.TabButtonHolder.Children)
+            {
+                if (!(bool)newValue)
+                {
+                    item.GestureRecognizers.Clear();
+                }
+                else
+                {
+                    var tapGestureRecognizer = new TapGestureRecognizer();
+                    tapGestureRecognizer.Tapped += segmentedControl.OnSegmentedItemTapped;
+
+                    if (!item.GestureRecognizers.Contains(tapGestureRecognizer))
+                        item.GestureRecognizers.Add(tapGestureRecognizer);
+                }
+            }
+        }
+    }
+
+    /// <summary>
+    /// Invoked when the color theme of the segmented control changes.
+    /// </summary>
+    private static void OnColorThemeChanged(BindableObject bindable, object oldValue, object newValue)
+    {
+        if (bindable is TMSegmentedControl segmentedControl)
+        {
+            foreach (var item in segmentedControl.SegmentedItems)
+            {
+                item.ColorTheme = segmentedControl.ColorTheme;
+            }
+        }
+    }
+    private static void OnSelectedIndexChanged(BindableObject bindable, object oldValue, object newValue)
+    {
+        (bindable as TMSegmentedControl)?.UpdateSelectedIndex();
+    }
+
+    /// <summary>
+    /// Invoked when the items source of the segmented control changes.
+    /// </summary>
+    private static void OnItemSourceChanged(BindableObject bindable, object oldValue, object newValue)
+    {
+        var segmentedControl = bindable as TMSegmentedControl;
+        segmentedControl.ClearSegmentedItems();
+        if (newValue == null)
+        {
+            return;
+        }
+        if (oldValue is INotifyCollectionChanged oldCollection)
+        {
+            oldCollection.CollectionChanged -= segmentedControl.OnItemsSourceCollectionChanged;
+        }
+        if (newValue is INotifyCollectionChanged newCollection)
+        {
+            newCollection.CollectionChanged += segmentedControl.OnItemsSourceCollectionChanged;
+        }
+        segmentedControl.PopulateItems();
+        segmentedControl.UpdateIsEnabled();
+        segmentedControl.UpdateSelectedIndex();
+    }
+    #endregion
+
+    #region Constructor
+    public TMSegmentedControl()
+    {
+        InitializeComponent();
+        UpdateIsEnabled();
+    }
+
+    #endregion
+    #region Private methods
+    /// <summary>
+    /// Populate Segment items in items source in the group
+    /// </summary>
+    private void PopulateItems()
+    {
+        if (ItemsSource == null)
+            return;
```

Comment:
When setting ItemsSource = null the Segment items will be cleared since there is a `ClearSegmentedItems` before this in the calling function. 
But segment items will remain the same when we use the method `ItemsSource.Clear()`, I have handled that now.